File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;
        int minprofit=prices[0];
        int maxprofit=0;
        for(int i=0;i<n;i++){
            int cost=prices[i]-minprofit;
            maxprofit=Math.max(maxprofit,cost);
            minprofit=Math.min(minprofit,prices[i]);
        }
        return maxprofit;
        
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cs
public class Solution {
    public int MaxProfit(int[] prices) {
      int max=0;
      int n=prices[prices.Length-1];
      for(int i=prices.Length-2;i>=0;i--){
        int price=0;
           if(prices[i]>n)n=prices[i];
           
           else{
             price=n-prices[i];
           }
           max=Math.Max(price,max);
      }
      return max;
    }
}